### DISCUSSION TITLE
ROBB Zigbee 8 button wall switch (ROB_200-007-0) not always working

### DISCUSSION THREAD

**Discussion by MarcoMiltenburg** [Koenkk/zigbee2mqtt#23838] [Devices]
Some time ago I bought two ROBB Zigbee 8 button wall switches (model [ROB_200-007-0](https://www.zigbee2mqtt.io/devices/ROB_200-007-0.html)) and I got them connected to Home Assistent with Zigbee2MQTT. They are recognised fine and although they do work, they don't work all the time.

It seems that if I don't use the switch for a certain time, the first time I press any of the buttons it's not seen by Zigbee2MQTT. When pressing another time or sometimes even a third time the button works. After that, usually it continues to work fine all the time. Until I haven't pressed any button for a lengthy time and then the same problems occurs again.

It's almost as if the device is falling asleep. But that is very unlikely as the green LED on the front of the switch always lights up. This indicates that the device is sending a message (I assume). However, nothing is shown in the Zigbee2MQTT log and no action is triggered in HA. Until I press another time, again the LED on the front lights up, and then it works fine and I see the log message in Zigbee2MQTT.

I've already paired the devices several times but this doesn't help. Problem remains the same. It also doesn't matter how close the ROBB switch is to my Zigbee antenna. Even if it's within 30 cm of the antenna it's not working the first button press. I'm using a Sonoff Zigbee 3.0 Dongle Plus-E.

What could this possible be?  And how to fix so that it always works, also on the first button press.

**Comment by MarcoMiltenburg** [comment]
Anybody using this device without any problems?  What firmware version do the switches have?

**Reply by TheYenZen** [reply]
Hi Marco, 
I have the same problem, same Sonoff Zigbee 3.0 Dongle Plus-E, and also tried several things. But nothing that helps 🙁.
As long as the switch is 'awake' it works fine. This however is not going to work for the long run, other members in the house are not amused anymore. So i might start looking for another switch....

**Reply by MarcoMiltenburg** [reply]
I'm not sure that happened but all of a sudden a couple of months ago it started working for both of my ROBB switches. Maybe there has been an update of Zigbee2MQTT or HA that fixed the issue. As far as I know I did not do an update of the firmware of the ROBB switch as I've never seen a firmware update for it.

**Reply by TheYenZen** [reply]
Did you use a blueprint to configure the switch?  I use a blueprint from FunkeyFlo and had to change the mode to queue to get dimming working. Maybe you have done something simular? Is recently updates HA and Z2MQTT but no change for me here

**Reply by mvannoorden** [reply]
I am curious how you got the device working at the first place. Got mine connected through Zigbee2MQTT, they are only showing up as 'battery' and tried several blueprints without succes

**Reply by TheYenZen** [reply]
@mvannoorden , I did got it working using a Blueprint for Z2MQTT, https://community.home-assistant.io/t/z2m-robb-smart-8-button-switch-rob-200-007-0-blueprint/441338
But i still have the issue that I have to 'wake' the switch when not used for a while (more then a hour I guess). so I have to click the same button twice the first time, after that it reacts on the first click.

**Reply by MarcoMiltenburg** [reply]
I'm not using a blueprint. I've added them manually to Zigbee2MQTT . From others I've understood that the following setting matters: Zigbee2MQTT ->Settings ->Home Assistant integration ->Home Assistant legacy action sensors. I have that disabled although I do make use of the "off_1" and "on_1" actions (and others) to trigger automations. Seems a bit counterintuitive.

**Reply by TheYenZen** [reply]
I had to change the setting "Home Assistant legacy action sensors"  to make it work again after the breaking update. before and after the update the switch does it's thing, but with the 'waking up' effect. I only have one switch so it's not easy to change this to test the other method (no blueprint).

**Comment by manuelkue** [comment]
> [...] However, nothing is shown in the Zigbee2MQTT log and no action is triggered in HA [...]

Could you try looking directly at the mqtt communication to see if any event is sent at all? If that's the case the problem lies in the automation - possibly after the breaking z2m 2.0 change, which needs other event handling if you went for the _action entities befor.

Follow the screenshots in this thread to reach the place where you can listen to mqtt topics: https://community.home-assistant.io/t/how-to-listen-to-mqtt-topic-under-dev-tools/422762

Under "Listen to a topic" enter `zigbee2mqtt/#`. Then press `Start listening`

Now you should see all zigbee events appear as they happen in your network. Press the buttons and look what kind of event is triggered / if an event is triggered.

Sidenode: The `/#` subscribes to all topics that are present in your mqtt under the `zigbee2mqtt` topic. To only listen for special topcis you can then provide a more granular topic afterwards like `zigbee2mqtt/<FRIENDLYNAME>/#`. This would subscribe to all zigbee events of your `<FRIENDLYNAME>`-device.

Read more here: https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/
Using mqtt for creating automations is quite flexible and might be an alternative to using Homeassistant entities.

**Reply by TheYenZen** [reply]
I have checked this and the first click does not trigger an event, so you could think that the switch itself is sleeping, then at the second click I see the events "on_1" pass by. I start to believe this has something to do with battery savings for the switch, not sure if this can be solved. maybe there's something to keep the switch awake some how

**Reply by manuelkue** [reply]
Mhm, when you mention the battery: Have you probably checked the battery / tested if replacing solves it?

**Comment by TheYenZen** [comment]
Yes, the battery is still 100%, the switch is just a couple of monts old. I did use a hue tap dial switch before this one, also battery powered, no issues there. But this one fit into the rest of the power sockets on the wall. Verzonden vanaf mijn Galaxy
-------- Oorspronkelijk bericht --------Van: manuelkue ***@***.***> Datum: 03-02-2025  17:28  (GMT+01:00) Aan: Koenkk/zigbee2mqtt ***@***.***> Cc: TheYenZen ***@***.***>, Comment ***@***.***> Onderwerp: Re: [Koenkk/zigbee2mqtt] ROBB Zigbee 8 button wall switch (ROB_200-007-0) not always working (Discussion #23838) 
Mhmm, have you checked the battery? :D Probably the voltage is too low after these months of usage?
